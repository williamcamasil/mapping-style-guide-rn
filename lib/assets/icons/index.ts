export * as Large from './large';
export * as Heavy from './heavy';
export * as Thin from './thin';
export * as Small from './small';
export * as Default from './default';
