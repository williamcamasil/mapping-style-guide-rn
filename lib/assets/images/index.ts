export { default as BtnTakePicture } from './BtnTakePicture';
export { default as BulletPoint } from './BulletPoint';
export { default as CheckboxImg } from './CheckboxImg';
export { default as DialogClose } from './DialogClose/index.svg';
export { default as Lock } from './Lock';
export { default as MaskedCardNumber } from './MaskedCardNumber';
export { default as Pointer } from './Pointer';
export { default as RadioImg } from './RadioImg';
