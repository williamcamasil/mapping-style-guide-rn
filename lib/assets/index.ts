export * as Icons from './icons';
export * as Images from './images';
export * as Lotties from './lotties';
