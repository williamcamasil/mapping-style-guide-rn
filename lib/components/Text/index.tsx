import React from 'react';
import {
  StyleProp, StyleSheet, Text as RNText, TextProps as RNTextProp, TextStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import {
  ColorPaletesType, FontWeightType, LineHeightType, Typography,
} from '../../tokens';
import {
  getColorStyle, getVariantStyle, getWeightStyle, TextVariantType,
} from './utils';

export type { TextVariantType } from './utils';

export type TextPropTypes = RNTextProp & {
  theme: AppThemeType;
  style?: StyleProp<TextStyle>;
  color?: ColorPaletesType;
  variant?: TextVariantType;
  weight?: FontWeightType;
  lineHeight?: LineHeightType;
};

const styles = StyleSheet.create({
  text: {
    letterSpacing: Typography.letterSpaces.default,
    flexShrink: 1,
  },
});

const Text: React.FC<TextPropTypes> = ({
  children,
  color,
  theme,
  style,
  variant = 'body',
  weight,
  lineHeight,
  allowFontScaling = false,
  ...others
}) => {

  const textStyles = useTextStyles(() => [
    styles.text,
    getColorStyle(theme, color ?? 'neutralGray600'),
    getVariantStyle(theme, variant, lineHeight),
    getWeightStyle(theme, weight),
    style,
  ], [color, lineHeight, style, theme, variant, weight]);

  return (
    <RNText
      {...others}
      allowFontScaling={allowFontScaling}
      style={textStyles}
    >
      {children}
    </RNText>
  );
};

export default withTheme(Text);
