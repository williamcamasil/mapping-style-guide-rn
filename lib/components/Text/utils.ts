import { TextStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType, FontWeightType, LineHeightType } from '../../tokens';
import { calcLineHeight } from '../../utils/typography';

export type TextVariantType =
  // heading
  'headingDisplay' |
  'headingXLarge' |
  'headingLarge' |
  'headingMedium' |
  'headingSmall' |
  // paragraph regular
  'subtitle' |
  'body' |
  'bodySmall' |
  'caption' |
  // small
  'small' |
  'smallCaps' |
  // one line
  'oneLineHuge' |
  'oneLineLarge' |
  'oneLineMedium' |
  'oneLineSmall' |
  'oneLineCaption' |
  'oneLineSemibold' |
  'oneLineBoldHuge' |
  'oneLineBoldExtraLarge' |
  'oneLineBoldLarge' |
  'oneLineBoldMedium' |
  'oneLineBoldSmall';

const getHeadingStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  headingDisplay: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.huge,
    lineHeight: calcLineHeight(theme.typography.sizes.huge, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingXLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXL, lineHeight ?? theme.typography.lineHeights.large),
  },
  headingMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.medium),
  },
});

const getSmallStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  small: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(14, lineHeight ?? theme.typography.lineHeights.medium),
  },
  smallCaps: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXS, lineHeight ?? theme.typography.lineHeights.medium),
    textTransform: 'uppercase' as TextStyle['textTransform'],
  },
});

const getOneLineStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  oneLineHuge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineLarge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineMedium: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineCaption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSemibold: {
    ...theme.typography.weights.semiBold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldHuge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldExtraLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
});

const getParagraphStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  subtitle: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.large),
  },
  bodySmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.large),
  },
  caption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.medium),
  },
  body: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.large),
  },
});

export function getVariantStyle(
  theme: AppThemeType,
  variant: TextVariantType,
  lineHeightName?: LineHeightType,
): TextStyle {
  const lineHeight = lineHeightName ? theme.typography.lineHeights[lineHeightName] : null;

  const variantStyles: Record<TextVariantType, TextStyle> = {
    // heading
    ...getHeadingStyles(theme, lineHeight),
    // small
    ...getSmallStyles(theme, lineHeight),
    // one line
    ...getOneLineStyles(theme, lineHeight),
    // paragraph
    ...getParagraphStyles(theme, lineHeight),
  };

  return variantStyles[variant];
}

export function getColorStyle(theme: AppThemeType, color: ColorPaletesType): TextStyle | null {
  return {
    color: theme.colors[color],
  };
}

export function getWeightStyle(theme: AppThemeType, weight?: FontWeightType): TextStyle | null {
  if (!weight) { return null; }

  return theme.typography.weights[weight];
}
