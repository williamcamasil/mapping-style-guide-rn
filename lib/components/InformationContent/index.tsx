import React from 'react';
import { ScrollView, StyleSheet } from 'react-native';

import { AppThemeType, withTheme } from '../../theme';
import Button from '../Button';
import Container from '../Container';
import Link from '../Link';
import Spacer from '../Spacer';
import Text from '../Text';

export type InformationContentPropsType = {
  title: string;
  description?: string;
  primaryButtonName?: string;
  header?: React.ReactNode;
  secondaryButtonName?: string;
  onPressPrimary?: () => void;
  onPressSecondary?: () => void;
  linkActionName?: string;
  onPressLinkAction?: () => void;
  theme: AppThemeType;
  children?: React.ReactNode;
};

const styles = StyleSheet.create({
  text: { textAlign: 'center' },
  container: { flexGrow: 1 },
});

const InformationContent: React.FC<InformationContentPropsType> = ({
  header,
  title,
  description,
  primaryButtonName,
  onPressPrimary,
  secondaryButtonName,
  onPressSecondary,
  linkActionName,
  onPressLinkAction,
  theme,
  children,
}) => {

  const renderDescription = () => {
    if (!children) return description ? <Text testID="information-content-description" style={styles.text} variant="body">{description}</Text> : null;
    return children;
  };

  return (
    <ScrollView
      showsVerticalScrollIndicator={false}
      alwaysBounceVertical={false}
      contentContainerStyle={styles.container}
    >
      <Container padding={theme.spacings.sLarge}>
        <Container justifyContent="center" alignItems="center">
          {header}
          <Spacer size={theme.spacings.sLarge} />
          <Text testID="information-content-title" style={styles.text} variant="headingSmall" color="neutralGray700">{title}</Text>
          <Spacer size={theme.spacings.sXXS} />
          {renderDescription()}
          <Spacer size={theme.spacings.sLarge} />
          {linkActionName ? <Link testID="information-content-link-action" onPress={onPressLinkAction}>{linkActionName}</Link> : null}
        </Container>
        {primaryButtonName ? <Button testID="information-content-primary-button" variant="containedPrimary" size="large" onPress={onPressPrimary}>{primaryButtonName}</Button> : null}
        {secondaryButtonName ? <Button testID="information-content-secondary-button" variant="text" size="large" onPress={onPressSecondary}>{secondaryButtonName}</Button> : null}
      </Container>
    </ScrollView>
  );
};

export default withTheme(InformationContent);
