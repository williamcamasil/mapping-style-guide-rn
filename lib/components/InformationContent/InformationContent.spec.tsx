import React from 'react';

import { render, fireEvent } from '@testing-library/react-native';

import InformationContent from '.';
import { Images } from '../../assets';
import Text from '../Text';

describe('InformationContent', () => {
  it('Should render InformationContent without Description', async () => {
    const result = render(<InformationContent
      header={<Images.DialogClose />}
      title="Title"
      primaryButtonName="Primary Button"
      secondaryButtonName="Secondary Button"
      onPressPrimary={jest.fn}
      onPressSecondary={jest.fn}
    />);

    expect(result.queryByTestId('information-content-description')).toBe(null);
  });

  it('Should render InformationContent whitout secondary button', async () => {
    const result = render(<InformationContent
      header={<Images.DialogClose />}
      title="Title"
      description="Description"
      primaryButtonName="Secondary Button"
      onPressPrimary={jest.fn}
    />);

    expect(result.queryByTestId('information-content-secondary-button')).toBe(null);

  });

  it('Should rcall function on button', async () => {
    const primaryButtonClick = jest.fn();
    const result = render(<InformationContent
      header={<Images.DialogClose />}
      title="Title"
      description="Description"
      primaryButtonName="Secondary Button"
      onPressPrimary={primaryButtonClick}
    />);

    fireEvent.press(result.getByTestId('information-content-primary-button'));

    expect(primaryButtonClick).toBeCalled();

  });

  it('Should render InformationContent with link action', async () => {
    const result = render(<InformationContent
      header={<Images.DialogClose />}
      title="Title"
      description="Description"
      linkActionName="Link action"
      onPressLinkAction={jest.fn}
    />);

    expect(result.queryByTestId('information-content-secondary-button')).toBe(null);

  });
});

describe('InformationContent snapshot', () => {
  it('default', () => {
    const infoStep = render(
      <InformationContent
        header={<Images.DialogClose />}
        title="Title"
        description="Description"
        primaryButtonName="Primary Button"
        secondaryButtonName="Secondary Button"
        onPressPrimary={jest.fn}
        onPressSecondary={jest.fn}
        linkActionName="Link action"
        onPressLinkAction={jest.fn}
      />,
    ).toJSON();

    expect(infoStep).toMatchSnapshot();
  });

  it('only primary button', () => {
    const infoStep = render(
      <InformationContent
        title="Title"
        primaryButtonName="Primary Button"
        onPressPrimary={jest.fn}
      />,
    ).toJSON();

    expect(infoStep).toMatchSnapshot();
  });

  it('only link action', () => {
    const infoStep = render(
      <InformationContent
        title="Title"
        linkActionName="Link action"
        onPressLinkAction={jest.fn}
      />,
    ).toJSON();

    expect(infoStep).toMatchSnapshot();
  });

  it('with Children', () => {
    const infoStep = render(
      <InformationContent
        title="Title"
        onPressLinkAction={jest.fn}
      >
        <Text variant="body">Mock children</Text>
      </InformationContent>,
    ).toJSON();

    expect(infoStep).toMatchSnapshot();
  });
});
