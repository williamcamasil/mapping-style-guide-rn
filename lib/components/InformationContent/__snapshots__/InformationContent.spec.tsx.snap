// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationContent snapshot default 1`] = `
<RCTScrollView
  alwaysBounceVertical={false}
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      pointerEvents="auto"
      style={
        {
          "alignItems": undefined,
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": undefined,
          "justifyContent": undefined,
          "padding": 28,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": undefined,
            "justifyContent": "center",
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
          }
        }
      >
        <SvgMock />
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#3E4149",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 23,
              "fontWeight": "bold",
              "letterSpacing": -0.02,
              "lineHeight": 28,
              "textAlign": "center",
            }
          }
          testID="information-content-title"
        >
          Title
        </Text>
        <Text
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
              "textAlign": "center",
            }
          }
          testID="information-content-description"
        >
          Description
        </Text>
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderBottomWidth": 1,
                "borderColor": "#0F4199",
                "opacity": 1,
                "paddingBottom": 8,
              }
            }
            testID="information-content-link-action"
          >
            <Text
              allowFontScaling={false}
              style={
                {
                  "color": "#0F4199",
                  "flexShrink": 1,
                  "fontFamily": "Averta CY",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "letterSpacing": -0.02,
                  "lineHeight": 20,
                }
              }
              testID="link-text"
            >
              Link action
            </Text>
          </View>
        </View>
      </View>
      <View
        style={{}}
        testID="container-button"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#013188",
              "borderRadius": 12,
              "borderWidth": 0,
              "display": "flex",
              "height": 64,
              "justifyContent": "center",
              "minWidth": 64,
              "opacity": 1,
              "paddingHorizontal": 32,
            }
          }
          testID="information-content-primary-button"
        >
          <Text
            allowFontScaling={false}
            style={
              {
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontFamily": "Averta CY",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": -0.02,
                "lineHeight": 19,
              }
            }
            testID="text-with-icon"
          >
            Primary Button
          </Text>
        </View>
      </View>
      <View
        style={{}}
        testID="container-button"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderRadius": 12,
              "borderWidth": 0,
              "display": "flex",
              "height": 64,
              "justifyContent": "center",
              "minWidth": 64,
              "opacity": 1,
              "paddingHorizontal": 32,
            }
          }
          testID="information-content-secondary-button"
        >
          <Text
            allowFontScaling={false}
            style={
              {
                "color": "#A4A7AF",
                "flexShrink": 1,
                "fontFamily": "Averta CY",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": -0.02,
                "lineHeight": 19,
              }
            }
            testID="text-with-icon"
          >
            Secondary Button
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`InformationContent snapshot only link action 1`] = `
<RCTScrollView
  alwaysBounceVertical={false}
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      pointerEvents="auto"
      style={
        {
          "alignItems": undefined,
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": undefined,
          "justifyContent": undefined,
          "padding": 28,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": undefined,
            "justifyContent": "center",
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
          }
        }
      >
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#3E4149",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 23,
              "fontWeight": "bold",
              "letterSpacing": -0.02,
              "lineHeight": 28,
              "textAlign": "center",
            }
          }
          testID="information-content-title"
        >
          Title
        </Text>
        <Text
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderBottomWidth": 1,
                "borderColor": "#0F4199",
                "opacity": 1,
                "paddingBottom": 8,
              }
            }
            testID="information-content-link-action"
          >
            <Text
              allowFontScaling={false}
              style={
                {
                  "color": "#0F4199",
                  "flexShrink": 1,
                  "fontFamily": "Averta CY",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "letterSpacing": -0.02,
                  "lineHeight": 20,
                }
              }
              testID="link-text"
            >
              Link action
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`InformationContent snapshot only primary button 1`] = `
<RCTScrollView
  alwaysBounceVertical={false}
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      pointerEvents="auto"
      style={
        {
          "alignItems": undefined,
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": undefined,
          "justifyContent": undefined,
          "padding": 28,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": undefined,
            "justifyContent": "center",
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
          }
        }
      >
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#3E4149",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 23,
              "fontWeight": "bold",
              "letterSpacing": -0.02,
              "lineHeight": 28,
              "textAlign": "center",
            }
          }
          testID="information-content-title"
        >
          Title
        </Text>
        <Text
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
      </View>
      <View
        style={{}}
        testID="container-button"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#013188",
              "borderRadius": 12,
              "borderWidth": 0,
              "display": "flex",
              "height": 64,
              "justifyContent": "center",
              "minWidth": 64,
              "opacity": 1,
              "paddingHorizontal": 32,
            }
          }
          testID="information-content-primary-button"
        >
          <Text
            allowFontScaling={false}
            style={
              {
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontFamily": "Averta CY",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": -0.02,
                "lineHeight": 19,
              }
            }
            testID="text-with-icon"
          >
            Primary Button
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`InformationContent snapshot with Children 1`] = `
<RCTScrollView
  alwaysBounceVertical={false}
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      pointerEvents="auto"
      style={
        {
          "alignItems": undefined,
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": undefined,
          "justifyContent": undefined,
          "padding": 28,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": undefined,
            "justifyContent": "center",
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "paddingVertical": undefined,
          }
        }
      >
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#3E4149",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 23,
              "fontWeight": "bold",
              "letterSpacing": -0.02,
              "lineHeight": 28,
              "textAlign": "center",
            }
          }
          testID="information-content-title"
        >
          Title
        </Text>
        <Text
          style={
            {
              "height": 12,
              "width": 12,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Mock children
        </Text>
        <Text
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
